code/game/machinery/stasis.dm

```diff
@@ -50,13 +80,51 @@
 	update_appearance()
 	return CLICK_ACTION_SUCCESS
 
+/// Toggle the filtering function on or off. This makes a bunch of scary-ass needles poke out of the bed.
+/// In exchange for having a passive function on hand to filter people's blood, it causes a little bit of damage.
+/// This damage only occurs upon initial activation, either through filtering starting by you being buckled or
+/// someone activating filtering on the bed you're buckled in already. Malicious doctors or silicons could toggle
+/// this over and over while you're helplessly buckled to the bed, though the effect of you being mutilated with
+/// needles will be obvious to onlookers.
+/obj/machinery/stasis/CtrlClick(mob/user)
+	if(!can_interact(user))
+		return
+	. = ..()
+	if(world.time < can_toggle)
+		return CLICK_ACTION_BLOCKING
+	filtering_enabled = !filtering_enabled
+	can_toggle = world.time + STASIS_TOGGLE_COOLDOWN
+	if(filtering_enabled)
+		var/sound/toggle_sound = sound('sound/items/unsheath.ogg')
+		toggle_sound.pitch = 2
+		playsound(src, toggle_sound, 60, TRUE,)
+	else
+		var/sound/toggle_sound = sound('sound/items/sheath.ogg')
+		toggle_sound.pitch = 2
+		playsound(src, toggle_sound, 60, TRUE,)
+	user.visible_message(span_notice("\The [src] filtering function [filtering_enabled ? "powers on. Thousands of micro-needles jut up from the mattress" : "shuts down. The micro-needles shunt back into the mattress invisibly"]."), \
+				span_notice("You [filtering_enabled ? "power on" : "shut down"] \the [src] filtering function. [filtering_enabled ? "Thousands of micro-needles jut up from the mattress" : "The micro-needles collapse back into the mattress, invisible"]."), \
+				span_hear("You hear [filtering_enabled ? " the unnerving sound of sharpened metal sliding against metal" : " a clunk of something metal"]."))
```

Comment:
```suggestion
	user.visible_message(
		span_notice("[src] filtering function [filtering_enabled ? "powers on. Thousands of micro-needles jut up from the mattress" : "shuts down. The micro-needles shunt back into the mattress invisibly"]."),
		span_notice("You [filtering_enabled ? "power on" : "shut down"] [src] filtering function. [filtering_enabled ? "Thousands of micro-needles jut up from the mattress" : "The micro-needles collapse back into the mattress, invisible"]."), 
		span_hear("You hear [filtering_enabled ? " the unnerving sound of sharpened metal sliding against metal" : " a clunk of something metal"]."),
	)
```
`\The` is unnecessary in `\The [src]` also we can multiline this in a way easier to read